mods/sethome/init.lua

```diff
@@ -24,42 +25,58 @@ loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
+function sethome.go_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		-- just a check to prevent the server crashing
+		return false
+	end
+	if homepos[name] then
+		player:setpos(homepos[name])
+		minetest.chat_send_player(name, "Teleported to home!")
+		return true
+	else
+		minetest.chat_send_player(name, "Set a home using /sethome")
+		return false
+	end
+end
 
 minetest.register_chatcommand("home", {
     description = "Teleport you to your home point",
     privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+    func = sethome.go_home,
 })
 
+function sethome.set_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		return false
+	end
+	local pos = player:getpos()
+	homepos[name] = pos
+	minetest.chat_send_player(name, "Home set!")
+
+	local output = io.open(homes_file, "w")
+	if output then
+		for i, v in pairs(homepos) do
+			output:write(v.x.." "..v.y.." "..v.z.." "..i.."\n")
+		end
+		io.close(output)
+	else
+		minetest.log("warning", "Couldn't open sethome's home file for saving positions : " .. homes_file)
+	end
+	return true
+end
+
 minetest.register_chatcommand("sethome", {
     description = "Set your home point",
     privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        local pos = player:getpos()
-        homepos[player:get_player_name()] = pos
-        minetest.chat_send_player(name, "Home set!")
-        changed = true
-        if changed then
-        	local output = io.open(homes_file, "w")
-            for i, v in pairs(homepos) do
-                output:write(v.x.." "..v.y.." "..v.z.." "..i.."\n")
-            end
-            io.close(output)
-            changed = false
-        end
-    end,
+    func = sethome.set_home,
 })
+
+function sethome.get_home(name)
+	if not name then
+		return nil
+	end
```

Comment:
The name isn't allowed to be `nil`, if it's passed there's been an error.  You should remove this check.